### ISSUE TITLE
Bug: Deploy Regression in v1.77.0 - Error: Waiter ChangeSetCreateComplete failed: Max attempts exceeded

### ISSUE THREAD

**Issue by rogerchi** [aws/aws-sam-cli#4906]
### Description:
Stacks that deployed without error with v1.76.0 are now failing with an error:
```
Waiting for changeset to be created..
Error: Waiter ChangeSetCreateComplete failed: Max attempts exceeded
```

### Steps to reproduce:
For us, we run SAM deploy as a CircleCI task -- steps to reproduce are to attempt to deploy an existing stack with sam cli.

### Observed result:
Changeset is created, but sam cli fails with the error above and the changeset is never executed.  If we pin the sam version to v1.76.0, the deployment succeeds.

### Expected result:
Changeset is applied and stack updates without error

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: linux -- circleci runner
2. `sam --version`: 1.77.0
3. AWS region: us-east-1

**Comment by moelasmar** [comment]
thanks @rogerchi for raising this issue. I am going to try reproduce this issue, but meanwhile can you give us an idea of the type of resources and number of resources that exist in your template

**Comment by rogerchi** [comment]
> can you give us an idea of the type of resources and number of resources that exist in your template

This has happened so for to two different stacks that we have -- one stack has 198 resources and the other has 217 (according to the CFN console).  The resources are a mixture of AppSync, Lambda Functions, SQS queues, etc. standard serverless stack stuff.  Pinning the circleci sam orb to v1.76.0 resolves the issue for us.

**Comment by moelasmar** [comment]
is it possible to run `sam deploy` with `--debug` option, and share the full logs with us ?

**Comment by rogerchi** [comment]
> is it possible to run `sam deploy` with `--debug` option, and share the full logs with us ?

Yes -- will be able to provide that later this afternoon.

**Comment by moelasmar** [comment]
can you try setting the environment variable `DEFAULT_POLL_DELAY` to 5 and rerun the deploy command and let me know if it fixes the issue or not?

``` export DEFAULT_POLL_DELAY=5```

**Comment by memoev** [comment]
Running into the same issue. Same circumstances, we're using `circleci/aws-sam-serverless@3.3.0` Circle CI [orb](https://circleci.com/developer/orbs/orb/circleci/aws-sam-serverless?version=3.3.0), which pulls latest version `1.77.0` and is causing all of our stack deploys (+150 resources) to fail. Current work around is explicitly setting `sam` to v1.76.0 like @rogerchi pointed out.

Will attempt using `export DEFAULT_POLL_DELAY=5` with v1.77.0.

**Comment by eankrame** [comment]
@moelasmar Did you mean to say set the environment variable `SAM_CLI_POLL_DELAY`? Looks like that's what [os.env is searching for](https://github.com/aws/aws-sam-cli/blob/develop/samcli/commands/deploy/command.py#L328)

**Comment by qingchm** [comment]
This is fixed by the latest release [1.78.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.78.0) Please have a look! Feel free to re-open if there are other issues regarding this!
